astropy/coordinates/earth.py

```diff
@@ -679,7 +686,17 @@ def get_itrs(self, obstime=None):
         # do this here to prevent a series of complicated circular imports
         from .builtin_frames import ITRS
 
-        return ITRS(x=self.x, y=self.y, z=self.z, obstime=obstime)
+        if location is None:
+            # No location provided, return geocentric ITRS coordinates
+            return ITRS(x=self.x, y=self.y, z=self.z, obstime=obstime)
+        else:
+            # Calculate relative Cartesian coordinates
+            itrs_vec = self.get_itrs(obstime).cartesian - location.get_itrs(obstime).cartesian
```

Comment:
`get_itrs` just wraps the x,y,z of `EarthLocation` in a new class with the same X, Y, Z, so the relative vector can be calculated more directly with
```
return ITRS(self.x - location.x, self.y - location.y, self.z - location.z, copy=False, obstime=obstime, location=location)
```